staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper_test.go

```diff
@@ -243,142 +285,348 @@ func TestRoundTripAndNewConnection(t *testing.T) {
+
+				ts, err := socks5.New(conf)
 				if err != nil {
-					t.Fatalf("%s: Error creating request: %s", k, err)
+					t.Errorf("failed to create sock5 server: %v", err)
 				}
+				return ts
+			}
+
+			testCases := map[string]struct {
+				proxyAuth              *url.Userinfo
+				serverConnectionHeader string
+				serverStatusCode       int
+				shouldError            bool
+			}{
+				"proxied tcp->tcp via SOCKS5": {
+					serverConnectionHeader: "Upgrade",
+					serverStatusCode:       http.StatusSwitchingProtocols,
+					shouldError:            false,
+				},
+				"proxied tcp->tcp via SOCKS5 with invalid auth": {
+					proxyAuth:              url.UserPassword("invalid", "auth"),
+					serverConnectionHeader: "Upgrade",
+					serverStatusCode:       http.StatusSwitchingProtocols,
+					shouldError:            true,
+				},
+				"proxied tcp->tcp via SOCKS5 with auth": {
+					proxyAuth:              url.UserPassword("proxyuser", "proxypasswd"),
+					serverConnectionHeader: "Upgrade",
+					serverStatusCode:       http.StatusSwitchingProtocols,
+					shouldError:            false,
+				},
+			}
 
-				spdyTransport := NewRoundTripper(testCase.clientTLS, redirect, redirect)
+			for name, testCase := range testCases {
+				t.Run(name, func(t *testing.T) {
+					server := httptest.NewServer(serverHandler(
+						t, serverHandlerConfig{
+							shouldError:      testCase.shouldError,
+							statusCode:       testCase.serverStatusCode,
+							connectionHeader: testCase.serverConnectionHeader,
+							upgradeHeader:    "SPDY/3.1",
+						},
+					))
+					defer server.Close()
+
+					req, err := http.NewRequest("GET", server.URL, nil)
+					if err != nil {
+						t.Fatalf("error creating request: %s", err)
+					}
 
-				var proxierCalled bool
-				var proxyCalledWithHost string
-				var proxyCalledWithAuth bool
-				var proxyCalledWithAuthHeader string
-				if testCase.proxyServerFunc != nil {
-					proxyHandler := goproxy.NewProxyHttpServer()
+					spdyTransport := NewRoundTripper(nil, redirect, redirect)
 
-					proxyHandler.OnRequest().HandleConnectFunc(func(host string, ctx *goproxy.ProxyCtx) (*goproxy.ConnectAction, string) {
-						proxyCalledWithHost = host
+					var proxierCalled bool
+					var proxyCalledWithHost string
 
-						proxyAuthHeaderName := "Proxy-Authorization"
-						_, proxyCalledWithAuth = ctx.Req.Header[proxyAuthHeaderName]
-						proxyCalledWithAuthHeader = ctx.Req.Header.Get(proxyAuthHeaderName)
-						return goproxy.OkConnect, host
-					})
+					interceptor := &Interceptor{proxyCalledWithHost: &proxyCalledWithHost}
 
-					proxy := testCase.proxyServerFunc(proxyHandler)
+					proxyHandler := socks5Server(nil, interceptor)
 
-					spdyTransport.proxier = func(proxierReq *http.Request) (*url.URL, error) {
-						proxierCalled = true
-						proxyURL, err := url.Parse(proxy.URL)
+					if testCase.proxyAuth != nil {
+						proxyHandler = socks5Server(&socks5.StaticCredentials{
+							"proxyuser": "proxypasswd",
```

Comment:
is there a reason these are hardcoded instead of being pulled from testCase.proxyAuth?

Reply:
This is the server credentials when we instantiate it, that's why they are hardcoded here, the tests then have their own credentials based on the test use-case (if we want to provide valid or invalid credentials) so I think this is correct.